uvtools/plot.py

```diff
@@ -466,3 +469,265 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
         images.append(imageio.imread(filename))
     imageio.mimsave(name, images)
     
+def plot_diff_waterfall(uvd1, uvd2, antpairpol, plot_type="all", 
+                        skip_check=False, save_path=None):
+    """Produce waterfall plot(s) of differenced visibilities.
+
+    Parameters
+    ----------
+    uvd1, uvd2 : pyuvdata.UVData
+        UVData objects which store visibilities to be differenced and their
+        associated metadata. They should have the same number of frequencies,
+        same baselines, and same times as each other.
```

Comment:
If you want to be fancy, it's possible to use the `UVData.select()` method to obtain matching shapes for the two objects. You just need to call `numpy.intersect1d` on the various metadata arrays to see which elements are common to both UVData objects.

Reply:
Hm. It looks like `numpy.intersect1d` is pretty conservative when making comparisons, only saying things intersect if the values are equal to one part in ~10^16. I suspect that if I were to use `numpy.intersect1d` in conjunction with `UVData.select()` to make things have the same shape, then it would often be the case that this ends up not plotting anything at all. In fact, the `check_metadata` conditions are much more lenient than this, just looking to see that the time arrays agree to a mean integration time and the frequency arrays agree to a mean channel width. (The check on the baseline vectors is a bit more stringent, using the default tolerance of `numpy.isclose`.)

Reply:
OK, sounds fine to me in that case.